packages/components/src/utils/users.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {UserProfile} from '@mattermost/types/users';
+
+import Preferences from '../constants/preferences';
+
+import {localizeMessage} from './i18n';
+
+export function getFullName(user: UserProfile): string {
+    if (user.first_name && user.last_name) {
+        return user.first_name + ' ' + user.last_name;
+    } else if (user.first_name) {
+        return user.first_name;
+    } else if (user.last_name) {
+        return user.last_name;
+    }
+
+    return '';
+}
+
+export function displayUsername(
+    user: UserProfile | null | undefined,
+    teammateNameDisplay: string,
+    useFallbackUsername = true,
+): string {
+    let name = useFallbackUsername ? localizeMessage('channel_loader.someone', 'Someone') : '';
```

Comment:
Instead of doing this in `displayUsername`, you could use `useIntl` in the component to do this translation. That'd also save you from having to make this package aware of how i18n works.

Plus, then we'd get rid of some usage of `localizeMessage`